Adafruit_IO/client.py

```diff
@@ -231,17 +243,52 @@ def receive_previous(self, feed):
         path = "feeds/{0}/data/previous".format(feed)
         return Data.from_dict(self._get(path))
 
-    def data(self, feed, data_id=None):
+    def data(self, feed, data_id=None, max_results=API_PAGE_LIMIT):
         """Retrieve data from a feed. If data_id is not specified then all the data
         for the feed will be returned in an array.
         :param string feed: Name/Key/ID of Adafruit IO feed.
         :param string data_id: ID of the piece of data to delete.
+        :param int max_results: The maximum number of results to return. To
+            return all data, set to None.
         """
-        if data_id is None:
-            path = "feeds/{0}/data".format(feed)
-            return list(map(Data.from_dict, self._get(path)))
-        path = "feeds/{0}/data/{1}".format(feed, data_id)
-        return Data.from_dict(self._get(path))
+        if data_id:
+            path = "feeds/{0}/data/{1}".format(feed, data_id)
+            return Data.from_dict(self._get(path))
+
+        params = {'limit': max_results} if max_results else None
+        data = []
+        path = "feeds/{0}/data".format(feed)
+        while True:
```

Comment:
Could you change this from `while True` to a non-blocking loop?

Reply:
I understand your distaste for `while True`. I implemented it this way in the event the user wants all data points. In that case it runs indefinitely until it runs out of data to retrieve. To remove the `while True`, I see at least two options. Either way would require an additional query to get the number of data points on a feed using feeds/{feed_key}/details query.

1. Disallow None for `max_results`. If we do this, we should expose the number of data points in a feed so that the user can request all data without having to guess how many points. Right now the `Feed` object doesn't include this.
2. Allow None as max_results but internally we query the feed details to get the total number of data points on the feed.

I can see pros/cons either way. Let me know if you have a preference.

Reply:
As a side note, it might be worth thinking about implementing more details for `Feed` anyway. Maybe I'll open another issue for discussing this.